actors/builtin/market/market_test.go

```diff
@@ -192,6 +192,32 @@ func TestMarketActor(t *testing.T) {
 			rt.Verify()
 		})
 
+		t.Run("fails if withdraw from non provider funds is not initiated by the recipient", func(t *testing.T) {
+			rt, actor := basicMarketSetup(t, owner, provider, worker, client)
+			actor.addParticipantFunds(rt, client, abi.NewTokenAmount(20))
+
+			rt.GetState(&st)
+			assert.Equal(t, abi.NewTokenAmount(20), actor.getEscrowBalance(rt, client))
+
+			rt.ExpectValidateCallerType(builtin.CallerTypesSignable...)
+			rt.ExpectValidateCallerAddr(client)
+			params := market.WithdrawBalanceParams{
+				ProviderOrClientAddress: client,
+				Amount:                  abi.NewTokenAmount(1),
+			}
+
+			// caller is not the recipient
+			rt.SetCaller(tutil.NewIDAddr(t, 909), builtin.AccountActorCodeID)
+			rt.ExpectAbort(exitcode.ErrForbidden, func() {
+				rt.Call(actor.WithdrawBalance, &params)
+			})
+			rt.Verify()
+
+			// verify there was no withdrawal
+			rt.GetState(&st)
+			assert.Equal(t, abi.NewTokenAmount(20), actor.getEscrowBalance(rt, client))
+		})
```

Comment:
Test also that withdraw from a miner cannot be triggered by account other than worker/owner.

Reply:
This has been added.